app/forms/AssociateUcr.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.AssociateKind.{Ducr, Mucr}
+import play.api.data.Forms._
+import play.api.data.format.{Formats, Formatter}
+import play.api.data.{Form, FormError, Forms}
+import play.api.libs.json.{Format, JsString, Json, JsonValidationError, Reads, Writes}
+import utils.validators.forms.FieldValidator._
+
+sealed abstract class AssociateKind(val formValue: String)
+
+object AssociateKind {
+  case object Mucr extends AssociateKind("mucr")
+  case object Ducr extends AssociateKind("ducr")
+
+  private val lookup = PartialFunction[String, AssociateKind] {
+    case Mucr.formValue => Mucr
+    case Ducr.formValue => Ducr
+  }
+
+  implicit val formatter: Formatter[AssociateKind] = new Formatter[AssociateKind] {
```

Comment:
Whats this for?

Reply:
Mapping

Reply:
Mapping what?

Reply:
without `Formatter` i am unable to use this class as part of mapping of `AssociateUcr`
```
"kind" -> of[AssociateKind],
```